java/adapter/jdbc/src/main/java/org/apache/arrow/adapter/jdbc/consumer/DateConsumer.java

```diff
@@ -89,7 +89,7 @@ public void consume(ResultSet resultSet) throws SQLException {
         if (day < 0 || day > MAX_DAY) {
           throw new IllegalArgumentException("Day overflow: " + day);
         }
-        vector.setSafe(currentIndex, day);
+        vector.set(currentIndex, day);
```

Comment:
please add comments for each of the set methods to indicate why setSafe is not required.

Reply:
Comments added. Thank you for the good suggestion.